docs/upgrade-notes/update-cumulus_id-type-indexes-CUMULUS-3449.md

```diff
@@ -0,0 +1,176 @@
+
+The following updates are included:
+
+- Update some cumulus_id columns to BIGINT to allow future data grows and fix data type mismatch between primary key and foreign key. The columns are:
+  - executions.cumulus_id
+  - executions.parent_cumulus_id
+  - files.granule_cumulus_id
+  - granules_executions.granule_cumulus_id
+  - granules_executions.execution_cumulus_id
+  - pdrs.execution_cumulus_id
+
+- Update indexes to provide the best chances of joins between tables
+  - Change granules table unique constraint from
+    `granules_granule_id_collection_cumulus_id_unique UNIQUE (granule_id, collection_cumulus_id)`
+    to
+    `granules_collection_cumulus_id_granule_id_unique UNIQUE (collection_cumulus_id, granule_id)`
+  - Add indexes `granules_granule_id_index` and `granules_provider_collection_cumulus_id_granule_id_index`
+    to `granules` table
+
+The updates will be automatically created as part of the bootstrap lambda function on deployment of the data-persistence module.
+
+*In cases where the type update and indexes are already applied, the updates will have no effect.*
+
+## Apply the Changes in Production Environment
+
+In production and some UAT environments, the type updates and indexes must be applied manually since the commands can take significant
+amount of time. Since these particular ALTER TABLE commands require an EXCLUSIVE LOCK on the table, it is recommended that
+all database activity be quiesced. This means that Ingest and Archive and other Cumulus functions must be shutdown before and during these commands.
+
+The table below from LPDAAC provides the table sizes before and after `alter table` commands, and timings.
+
+| Table Name | Original Table Size | ALTER Run Time (clock) | New Table Size | Number of Rows |
+|---|---|---|---|---|
+| executions | 1.8 TB | 41 hours | 1.4 TB | 374,170,855 |
+| files | 1.2 TB | 7.5 hours | 850 GB | 1,129,847,659 |
+| granule_executions | 20 GB | 1 hours | 13 GB | 182,566,709 |
+
+## Tools Used
+
+Since the update commands can take several hours to run based on table size and IO throughput, it is recommended that the commands are run in an EC2 instance
+in the AWS environment in a tmux or screen session. This will minimize the number of network hops and potential disconnects between the database client
+and the database. In addition, this will allow operators applying the patch to check on progress periodically and not worry about credential expiration or
+other issues that would result in the client being killed.
+
+There are many available resources for tmux and the psql client. The following is for your referrence.
+
+- [PostgreSQL Documentation](https://www.postgresql.org/docs/13/app-psql.html)
+- [Tmux tutorial](https://www.linuxtrainingacademy.com/tmux-tutorial/)
+
+:::note
+The version 9.x psql will be installed. This will work for all our standard SQL but some meta-commands like \d may not work.
+:::
+
+Basic commands for running SQL commands
```

Comment:
Do these instructions (tmux commands in particular) work from an AWS console, or should users *only* use a console ssm session?